server/database/models/tournamentModel.ts

```diff
@@ -1,2 +1,57 @@
-// import mongoose from 'mongoose';
-// import Tournament from './../../../shared/types/Tournament';
+import mongoose, { Schema, Types, Model } from 'mongoose';
+import Tournament, { Match, Team } from '../../../shared/types/Tournament';
+import { TeamDocument } from './teamModel';
+
+const MatchSchema = new Schema({
+  rivals: {
+    teamA: { type: Types.ObjectId, ref: 'Team' },
+    teamB: { type: Types.ObjectId, ref: 'Team' },
+  },
+  score: {
+    teamA: Number,
+    teamB: Number,
+  },
```

Comment:
This model seems a bit disjointed. Is there any particular reason for such a design?

What do you think about this?
```suggestion
  participantA: {
    team: { type: Types.ObjectId, ref: 'Team' },
    score: Number,
  },
  participantB: {
    team: { type: Types.ObjectId, ref: 'Team' },
    score: Number,
  },
```

Reply:
Admit that. + I think better way to get ObjectId type is to get it from Schema (Schema.Types.ObjectId). We discussed it earlier.

Reply:
1. I agree (design) and I'm changing it to your suggestion @mblaze 
2. The difference between `Schema.Types.String and Types.String`
[https://github.com/Automattic/mongoose/issues/1671](url)
[https://stackoverflow.com/questions/65831345/what-is-the-difference-between-mongoose-objectid-and-mongoose-types-objectid](url)
[https://mongoosejs.com/docs/schematypes.html](url)

```
const schema = new Schema({
     _someId: mongoose.Schema.Types.ObjectId,
});
 
const Thing = mongoose.model('Thing', schema);

m._someId = new mongoose.Types.ObjectId;
```